.gitlab-ci.yml

```diff
@@ -0,0 +1,124 @@
+image: golang:1.11-alpine3.9
+
+cache:
+  paths:
+    - /go/src/github.com
+    - /go/src/golang.org
+    - /go/src/google.golang.org
+    - /go/src/gopkg.in
+
+variables:
+   DOCKER_HOST: tcp://docker:2375/
+   DOCKER_DRIVER: overlay2
+   DOCKER_REPOSITORY: mendersoftware/mender-artifact
+
+services:
+  - docker:dind
+
+stages:
+  - test
+  - build
+  - publish
+  - integration
+
+before_script:
+  - mkdir -p /go/src/github.com/mendersoftware /go/src/_/builds
+  - cp -r $CI_PROJECT_DIR /go/src/github.com/mendersoftware/mender-artifact
+  - ln -s /go/src/github.com/mendersoftware /go/src/_/builds/mendersoftware
+  - cd /go/src/github.com/mendersoftware/mender-artifact
+  - apk update && apk add git make
+  # Install code coverage tooling
+  - make get-tools
+
+test:static:
+  stage: test
+  script:
+    - apk add bash e2fsprogs e2fsprogs-extra gcc libc6-compat mtools musl-dev parted perl-utils xz-dev
+    - make extracheck
+    - make coverage
+    - mv /go/src/github.com/mendersoftware/mender-artifact/coverage.txt $CI_PROJECT_DIR/coverage.txt
+  artifacts:
+    untracked: true
+    paths:
+      - coverage.txt
+
+build:
```

Comment:
We can call this job `build:docker`, so that it is consistent with `build:make` below